sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRowToColumnarExec.scala

```diff
@@ -138,9 +138,24 @@ private[rapids] object GpuRowToColumnConverter {
           getConverterForType(v, vcn))
       case (NullType, true) =>
         NullConverter
+    }
+
+    val unknownMatch: PartialFunction[(DataType, Boolean), TypeConverter] = {
       case (unknown, _) => throw new UnsupportedOperationException(
         s"Type $unknown not supported")
     }
+
+    // get shim matches
+    val shimMatches = GpuTypeShims.getConverterForType
```

Comment:
This looks a little complicated.
You can create two APIs in `GpuTypeShims` (Maybe `RapidsTypeShim` would be better), called `isShimSupported(DataType): Boolean` and `getConverterForType(DataType, Boolean): Converter`, then just update the code here like
```diff
      case (NullType, true) =>
        NullConverter
+     case (otherType, nullable) if GpuTypeShims.isShimSupported(otherType) =>
+       GpuTypeShims.getConverterForType(otherType, nullable)
      case (unknown, _) => throw new UnsupportedOperationException(
        s"Type $unknown not supported")    
```

Reply:
Good comment,  updated.